std/wasi/snapshot_preview1.ts

```diff
@@ -228,9 +228,11 @@ const clock_time_monotonic = function (): bigint {
 const clock_time_process = clock_time_monotonic;
 const clock_time_thread = clock_time_monotonic;
 
-type Syscall = (...args: unknown[]) => number;
-function syscall(target: Syscall): Syscall {
-  return function (...args: unknown[]): number {
+// deno-lint-ignore no-explicit-any
+type Syscall<T extends any[]> = (...args: T) => number;
+// deno-lint-ignore no-explicit-any
+function syscall<T extends any[]>(target: Syscall<T>): Syscall<T> {
```

Comment:
Kinda pointless to change the interface AND still put a lint ignores isn't it?

Reply:
Yes, your thought is understandable. We could also add `// deno-lint-ignore ban-types` without changing interface.
But comparing with untyped `Function`, `(...args: T) => number` would be better in that it specifies return type to be number.